src/mstore.erl

```diff
@@ -566,17 +602,23 @@ do_write(M=#mfile{offset=Offset, size=S, file=F, index=Idx},
          DataSize, Metric, Position, Value) when
       is_binary(Metric),
       is_binary(Value) ->
+
+    %% As part of a write operation, both the index and mstore files may be
+    %% updated. There is no atomicity gaurantees, and a fault may occur after
+    %% one of the writes have been processed. Writing data to the index before
+    %% writing to the mstore ensures that offsets are calculated so as not to
+    %% cause data to overlap. In other words, data loss is acceptable, but not
+    %% data corruption.
     {M1, Base} =
         case btrie:find(Metric, Idx) of
             error ->
                 Pos = M#mfile.next,
                 Mx = M#mfile{next=Pos+1, index=btrie:store(Metric, Pos, Idx)},
-                file:write_file([M#mfile.name | ".idx"],
-                                <<(byte_size(Metric)):16/integer, Metric/binary>>,
-                                [read, append]),
+                Bin = <<(byte_size(Metric)):16/integer, Metric/binary>>,
+                ok = file:write_file([M#mfile.name | ".idx"], Bin, [read, append]),
```

Comment:
Line 618 is too long:                 ok = file:write_file([M#mfile.name | ".idx"], Bin, [read, append]),.